gobblin-data-management/src/main/java/org/apache/gobblin/data/management/retention/version/DatasetRetentionSummary.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.gobblin.data.management.retention.version;
+
+import lombok.Data;
+
+
+/**
+ * Aggregates information from {@link org.apache.gobblin.data.management.retention.DatasetCleaner} to
+ * track the number of versions selected for deletion for a dataset.
+ */
+@Data
+public class DatasetRetentionSummary {
+  private final String datasetUrn;
+  private final int versionsSelectedForDeletion;
+  private final boolean successfullyDeleted;
```

Comment:
in the impl, it appeared that `versionsSelectedForDeletion` is truly the total number successfully deleted.  e.g. is there any way that the number selected might not equal a *non-zero* number successfully deleted?

if not, I suggest renaming to `versionsDeleted` and reversing the boolean to `hadExecutionFailure`. the latter of course discerns between 0 found to delete and >0 found, but none actually successful.  calling out a tad more explicitly in javadoc seems worthwhile